src/main/scala/beam/agentsim/infrastructure/ChargingNetwork.scala

```diff
@@ -69,8 +78,11 @@ class ChargingNetwork(managerId: Id[VehicleManager], chargingStationsQTree: Quad
     */
   def attemptToConnectVehicle(tick: Int, vehicle: BeamVehicle, theSender: ActorRef): ChargingVehicle = {
     vehicle.stall match {
-      case Some(stall) => lookupStation(stall).map(station => station.connect(tick, vehicle, stall, theSender)).get
-      case _           => throw new RuntimeException(s"Vehicle $vehicle doesn't have a stall!")
+      case Some(stall) =>
+        lookupStation(stall.tazId, stall.parkingType, stall.chargingPointType.get)
+          .map(station => station.connect(tick, vehicle, stall, theSender))
+          .get
```

Comment:
is it possible to get a runtime exception here because there is no such station?

Reply:
Okay fixed!